torch_ttnn/passes/lowering/to_tt_pass.py

```diff
@@ -318,9 +318,11 @@ def torch_dtype_to_ttnn_dtype(dtype: torch.dtype):
 # RuntimeError: TT_FATAL @ ../tt_metal/impl/buffers/buffer.cpp:38: valid_page_size
 # For valid non-interleaved buffers page size 2048 must equal buffer size X. For interleaved-buffers page size should be divisible by buffer size
 def has_valid_page_size(shape, strict=False):
-    if len(shape) >= 2 and shape[-1] > 0:
-        return shape[-1] % 32 == 0 or (not strict and shape[-1] < 32)
-    return False
+    if len(shape) < 2 or shape[-1] == 0:
+        return False
+    if not strict and shape[-1] < 32:
+        return True
+    return shape[-2] % 32 == 0 and shape[-1] % 32 == 0
```

Comment:
I found that the previous strict condition is still not enough. The buffer size needs to be divisible by or equal to the page size 2048 by this assert https://github.com/tenstorrent/tt-metal/blob/7040c7836d2c2953971834981ff79ff21c0d36e5/tt_metal/impl/buffers/buffer.cpp#L33-L38

And even though `(16, 64)` bfloat16 tensor, which size = 2048, will pass assertions, it gives out wrong results (in `test_gt.py` tests) or memory corruputions (in tests add by this PR) during my testing. I assume it is due to [the only supported tile shape is 32x32](https://github.com/tenstorrent/tt-metal/blob/main/tech_reports/tensor_layouts/tensor_layouts.md#321-tile-shapes) currently?

Therefore, I add the condition here to check both dims must be divisible by 32.

Reply:
It now supports non-tiled size